Detectors/GlobalTrackingWorkflow/tofworkflow/src/tof-matcher-global.cxx

```diff
@@ -0,0 +1,165 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file   tof-reco-workflow.cxx
+/// @author Francesco Noferini
+/// @since  2019-05-22
+/// @brief  Basic DPL workflow for TOF reconstruction starting from digits
+
+#include "DetectorsBase/Propagator.h"
+#include "GlobalTrackingWorkflow/TrackTPCITSReaderSpec.h"
+#include "TOFWorkflow/ClusterReaderSpec.h"
+#include "TOFWorkflow/TOFMatchedWriterSpec.h"
+#include "TOFWorkflow/TOFCalibWriterSpec.h"
+#include "Framework/WorkflowSpec.h"
+#include "Framework/ConfigParamSpec.h"
+#include "TOFWorkflow/RecoWorkflowSpec.h"
+#include "Algorithm/RangeTokenizer.h"
+#include "FairLogger.h"
+#include "CommonUtils/ConfigurableParam.h"
+#include "DetectorsCommonDataFormats/NameConf.h"
+
+// GRP
+#include "DataFormatsParameters/GRPObject.h"
+
+// FIT
+#include "FT0Workflow/RecPointReaderSpec.h"
+
+#include <string>
+#include <stdexcept>
+#include <unordered_map>
+
+// add workflow options, note that customization needs to be declared before
+// including Framework/runDataProcessing
+void customize(std::vector<o2::framework::ConfigParamSpec>& workflowOptions)
+{
+  workflowOptions.push_back(ConfigParamSpec{"input-type", o2::framework::VariantType::String, "clusters,tracks", {"clusters, tracks, fit"}});
```

Comment:
why providing the clusters and tracks is an option? Don't you always need them?

Reply:
because in principle I can use the DPL in pipe with something which already provides them. In this way I can switchoff the readers.

Reply:
Ah, just noticed that you use to add the readers. But I don't think you can have e.g. clusters from the root file and tracks from other device. Perhaps you can use ``disable-root-input`` and ``disable-root-output``, already used by few other devices.

Reply:
I will do it in a next PR